homeassistant/components/sensor/comed_hourly_pricing.py

```diff
@@ -97,12 +97,12 @@ def update(self):
         try:
             if self.type == CONF_FIVE_MINUTE:
                 url_string = _RESOURCE + '?type=5minutefeed'
-                response = get(url_string, timeout=10)
+                response = get(url_string, timeout=60)
```

Comment:
This is too long. It means that we occupy a worker thread for 60 seconds! We should not wait this long.

Reply:
Are worker threads that expensive?  Waiting for the full timeout wouldn't occur that often; usually the API responds pretty quickly.  However, the times when it takes a while to respond are usually when electricity demand is high meaning the price has gone up significantly.  I would rather tie up a worker thread waiting for the response because my primary usage of this API is to trigger automations when the electricity price spikes.  What about making the timeout a configuration item that the user could adjust?

Reply:
Configuration option is not a good way to go, we should not allow the user to enter configuration that will slow/break his Home Assistant and then get him to blame HA is slow.

60 seconds is way too long. No server should keep requests waiting that long.

Reply:
What is a more reasonable timeout then? 30?

Reply:
I honestly think that 10 is a reasonable timeout. These responses are answered by a machine right 😆 

Do you have data on how long it takes to answer during peak hours?

Reply:
Ten seconds is a reasonable amount for a high-quality, responsive API.  Unfortunately, this API doesn't seem to handle high load very well.  During high load times, the API can typically take 15-30 seconds to respond, though I have seen it take 30-60 seconds occasionally.

Reply:
Alright, you can increase it to a 60 second timeout if you rewrite the code to use asyncio. That way you will not be occupying a worker.

Example asyncio code to make a request: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/hassio.py#L80-L97